### PR TITLE
Improves documentation in R/

### PR DESCRIPTION
**Pull request by nikolase90**
Looked over the following files:

`R/`
- [x] clustering.R
- [x] explanation.R
- [x] features.R
- [x] models.R
- [x] observations.R
- [x] plot.R
- [x] predictions.R
- [x] sampling.R
- [x] shapley.R
- [x] transformation.R
- [x] utils.R

I've also made some small improvements for files in `src`, but the remaining parts of that will be done in a different PR.

### CODE REVIEW

**Comment by nikolase90** [comment]
You don't have to look at the `*.Rd`. Those are automatically generated.

File: R/features.R [hunk 1]
```diff
@@ -1,24 +1,41 @@
...
 #'
-#' @inheritParams global_arguments
+#' @param m Positive integer. Total number of features.
+#' @param exact Logical. If \code{TRUE} all \code{2^m} combinations are generated, otherwise a
+#' subsample of the combinations is used.
+#' @param n_combinations Positive integer. Note that if \code{exact} equals \code{TRUE},
+#' \code{n_combinations} is ignored. However, if \code{m > 12} you'll need to add a positive integer
+#' value for \code{n_combinations}.
+#' @param shapley_weight_inf_replacement Numeric.
+#' @param reduce_dim Logical.
```
[the remark this answers is not in the document]
Comment (martinju) [reply R/features.R]: We may also consider removing this and just including a comment in the code similar to the above. I see no situations we you don't want to do this.

File: R/features.R [hunk 2]
```diff
@@ -1,24 +1,41 @@
...
-#' \item{ID}{Positive integer. Unique key for combination}
-#' \item{features}{List of integer vectors}
-#' \item{nfeatures}{Positive integer}
-#' \item{N}{Positive integer}
+#' \item{ID}{Positive integer. Represents a unique key for each combination. Note that the table
+#' is sorted by \code{ID}, so that is always equal to \code{x[["ID"]] = 1:nrow(x)}.}
+#' \item{features}{List. Each item of the list is an integer vector where \code{features[[i]]}
+#' represents the indices of the features included in combination \code{i}. Note that all the items
+#' are sorted such that \code{features[[i]] == sort(features[[i]])} is always true.}
+#' \item{nfeatures}{Positive integer. \code{nfeatures[i]} equals the number of features in combination
```
Comment (martinju) [inline R/features.R]: Change nfeatures to n_features?

File: R/features.R [hunk 3]
```diff
@@ -1,24 +1,41 @@
...
+#' # All combinations
+#' x <- feature_combinations(m = 5)
+#' nrow(x) # Equals 2^5 = 32
+#'
+#' # Subsample of combinations
+#' x <- feature_combinations(m = 13, n_combinations = 1e3)
 feature_combinations <- function(m, exact = TRUE, n_combinations = 200, shapley_weight_inf_replacement = 10^6, reduce_dim = TRUE) {
 
-  # Force user to use a natural number for n_combinations if m > 10
+  # Force user to use a natural number for n_combinations if m > 12
```
Comment (martinju) [inline R/features.R]: This is a comment for line 42-45:

"Due to computational complexity, we recommend not using exact = TRUE if the number of features is larger than 12. 
We recommend setting n_combinations = 10 000, or some larger number < 2^m. Note that you can force the use of the exact method (i.e. n_combinations = NULL) by setting n_combinations equal to 2^m, where m is the number of features."

File: R/models.R [hunk 4]
```diff
@@ -1,9 +1,10 @@
-#' Predict on vector form
+#' Generate predictions for different model classes
 #'
 #' @description Performs prediction of response \code{\link[stats]{lm}}, \code{\link[stats]{glm}},
-#' \code{\link[ranger]{ranger}} and \code{\link[xgboost]{xgboost}} with binary or continuous response.
+#' \code{\link[ranger]{ranger}} and \code{\link[xgboost]{xgboost}} with binary or continuous
+#' response. See details for more information.
 #'
-#' @param x Object of class inheriting from one of the supported models. See details.
+#' @param x Object of class inheriting from one of the supported models.
```
Comment (martinju) [inline R/models.R]: I don't understand the sentence. Different suggestion below.
```suggestion
#' @param x Model object for the model to be explained.
```

File: R/models.R [hunk 5]
```diff
@@ -114,28 +118,38 @@ predict_model.mgcv <- function(x, newdata) {
...
+#'
+#' @description The function checks whether the model given by \code{x} is
+#' supported, and if it is a regression- or a classification model. If \code{x} is
+#' not a supported model the function will return an error message, otherwise it will
+#' return either \code{"regression"} or \code{"classification"}.
+#'
+#' @inheritParams predict_model
+#'
+#' @details See \code{\link{predict_model}} for more information about
+#' what type of models \code{shapr} currently support.
```
Comment (martinju) [inline R/models.R]: Add: 
and the package vignette: \code{vignette("understanding_shapr", package = "shapr")} for how to use a custom model.

File: R/observations.R [hunk 6]
```diff
@@ -71,16 +89,18 @@ observation_impute <- function(W_kernel, S, x_train, x_test, w_threshold = .7, n
 
 #' Generate data used for predictions
 #'
+#' @param x Explainer object. See \code{\link{explain}} for more information.
+#'
 #' @param n_samples Positive integer. Indicating the maximum number of samples to use in the
 #' Monte Carlo integration for every conditional expectation.
 #'
 #' @param seed Positive integer. If \code{NULL} a random seed will be used.
```
Comment (martinju) [inline R/observations.R]:
```suggestion
#' @param seed Positive integer. If \code{NULL} the seed will be inherited from the calling environment.
```

File: R/observations.R [hunk 7]
```diff
@@ -71,16 +89,18 @@ observation_impute <- function(W_kernel, S, x_train, x_test, w_threshold = .7, n
...
 #'
 #' @param index_features Positive integer vector. Only used internally.
```
Comment (martinju) [inline R/observations.R]:
```suggestion
#' @param index_features Positive integer vector. Specifies combinations to apply to the present method. NULL means all. Only used internally.
```

Comment (nikolase90) [reply R/observations.R]: Fixed with 36bc100.

File: R/models.R (hunk 5 above)
Comment (nikolase90) [reply R/models.R]: We don't say anything about using a custom model in the vignette?

File: R/explanation.R [hunk 8]
```diff
@@ -12,22 +10,87 @@
...
+#'   \item{model}{Model object}
+#'   \item{p}{Numeric vector}
+#'   \item{x_test}{data.table}
+#' }
 #'
-#' @return data.frame. Contains the estimated Shapley values for the test data. Note that
-#' the dimensions of the data.frame equals \code{n x (p+1)}, where \code{n} equals the number
-#' of test observations, and \code{p} equals the total number of features.
+#' Note that the items \code{model}, \code{p} amd \code{x_test} is mostly addded due
+#' to the implementation of \code{plot.shapr}. If you'd like to look at the numerical results
```
[the remark this answers is not in the document]
Comment (nikolase90) [reply R/explanation.R]: Fixed with 183e0bc.

File: R/explanation.R [hunk 9]
```diff
@@ -66,24 +129,25 @@ explain <- function(x, explainer, approach, prediction_zero, ...) {
...
-#' and \code{type='AICc_each_k'} or \code{type='AICc_full'}. Maximum number of iterations when
-#' optimizing the AICc.
+#' @param eval_max_aicc Positive integer. Maximum number of iterations when
+#' optimizing the AICc. Note that this argument is only applicable when
+#' \code{approach = "empirical"}, and \code{type} is either equal to
+#' \code{"AICc_each_k"} or \code{"AICc_full"}
 #'
-#' @param start_aicc Numeric. Only applicable when \code{approach='empirical'} and
-#' \code{type='AICc_each_k'} or \code{type='AICc_full'}. Starting value when optimizing the AICc.
+#' @param start_aicc Numeric. Start value when optimizing the AICc. Note that this argument
```
[the remark this answers is not in the document]
Comment (nikolase90) [reply R/explanation.R]: Fixed with 1cf0cda.

File: R/explanation.R [hunk 10]
```diff
@@ -227,7 +287,26 @@ explain.combined <- function(x, explainer, approach, prediction_zero, mu = NULL,
 
 }
 
+#' Helper function used in \code{\link{explain.combined}}
+#'
+#' @param n_features Integer vector. Note that
+#' \code{length(n_combinations) <= 2^m}, where \code{m} equals the number
+#' of features.
```
[the remark this answers is not in the document]
Comment (nikolase90) [reply R/explanation.R]: There is a typo in this case. Fixed with 8550248.

File: R/explanation.R [hunk 11]
```diff
@@ -227,7 +287,26 @@ explain.combined <- function(x, explainer, approach, prediction_zero, mu = NULL,
...
+#' @param approach Character vector of length \code{m}. All elements should be
```
[the remark this answers is not in the document]
Comment (nikolase90) [reply R/explanation.R]: This will be incorrect. `n_features` is a vector, i.e. when `m = 3`, `n_features = c(0, 1, 1, 1, 2, 2, 2, 3)`. Makes sense?

File: R/shapley.R [hunk 12]
```diff
@@ -44,28 +48,75 @@ weight_matrix <- function(X, use_shapley_weights_in_W = TRUE, normalize_W_weight
...
 #'
 #' @param n_combinations Integer. The number of feature combinations to sample. If \code{NULL},
 #' the exact method is used and all combinations are considered. The maximum number of
 #' combinations equals \code{2^p}.
 #'
-#' @return A list to be used by \code{explain} to compute the kernel SHAP values (\code{Kshap}).
+#' @return Named list that contains the follwing items:
+#' \describe{
+#'   \item{exact}{Boolean. Equals \code{TRUE} if \code{n_combinations = NULL} or
+#'   \code{n_combinations < 2^p}, otherwise \code{FALSE}.}
```
Comment (martinju) [inline R/shapley.R]: p = n_features

File: R/transformation.R [hunk 13]
```diff
@@ -1,17 +1,20 @@
-#' Transforms new data to standardized normal (dimension 1) based on other data transformations
+#' Transforms new data to a standardized normal distribution
 #'
-#' Handled in this way in order to allow using the apply function over this function
-#' @param zx Vector where the first part is the Gaussian data, and last part is
-#' the data with the original transformation
-#' @param n_z How many elements of \code{zx} that belongs to the z-part (new data)
-#' @param type The quantile type used when back-transforming. 7 (default) is the default in \code{stats::quantile()}.
+#' @param zx Numeric vector. The first \code{n_z} items is the Gaussian data, and last part is
```
Comment (martinju) [inline R/transformation.R]:
```suggestion
#' @param zx Numeric vector. The first \code{n_z} items are the Gaussian data, and the last part is
```

File: R/transformation.R [hunk 14]
```diff
@@ -1,17 +1,20 @@
...
+#' the data with the original transformation.
+#' @param n_z Positive integer. Number of elements of \code{zx} that belongs to new data.
+#' @param type Integer between \code{1} and \code{9}. See \code{\link[stats]{quantile}} for more
```
Comment (martinju) [inline R/transformation.R]: Couldn't we just pass type as a ... to quantile? I don't think any user will ever care to change this setting.

File: R/transformation.R [hunk 15]
```diff
@@ -50,16 +55,17 @@ gaussian_transform_separate <- function(yx, n_y) {
   return(z_y)
 }
 
-#' Transforms a sample to standardized normal (dimension 1)
+#' Transforms a sample to standardized normal distribution
 #'
-#' @param x Vector of data to transform
+#' @param x Numeric vector
```
Comment (martinju) [inline R/transformation.R]:
```suggestion
#' @param x Numeric vector. The data to transform to standard normal.
```

**Comment by nikolase90** [comment]
> Note done yet, but here are the first half of the review
> 
> A few general comments in addition to the inline comments below:
> 
> * [ ]  I think we should run some spell checking. See for instance this package: https://github.com/ropensci/spelling. It also gives ability to run the spell checker as a unit test.
> * [ ]  I suggest we change `m` and `nfeatures` to `n_features` for consistency throughout the package.
> * [ ]  We should be better at specifying what is the default value of input parameters. At least I find that helpful  when reading documentations.

Agree to most of it. I'll do the spell check, but I think we should only add documentation changes in this PR. So we could streamline the naming of the variables in a different one (just to make sure that the actually code changes is clear). 

It is not necessary to specify the default parameters when documenting the actual parameters. The default values are shown in the `Usage` section, i.e. run `?shapr` to see an example.

File: R/zzz.R [hunk 16]
```diff
@@ -4,10 +4,6 @@
   if (getRversion() >= "2.15.1") {
```
Comment (martinju) [inline R/zzz.R]: Do we need this if test when we anyway require a newer R version?

File: src/AICc.cpp [hunk 17]
```diff
@@ -148,19 +148,18 @@ arma::vec aicc_full_single_cpp(arma::mat X, arma::mat mcov, bool S_scale_dist, d
...
-//' @param S_scale_dist logical indicating whether the Mahalanobis distance should be scaled with the number of variables
-//' @param y vector with the "response variable"
-//'
-//' @export
+//' @param h Numeric. Specifies the scaling (sigma)
+//' @param X_list List
+//' @param mcov_list List
+//' @param S_scale_dist Logical. Indicates whether Mahalanobis distance should be scaled with the
+//' number of variables
+//' @param y_list List.
```
Comment (martinju) [inline src/AICc.cpp]: Should document these properly.

File: src/distance.cpp [hunk 18]
```diff
@@ -11,6 +9,8 @@ using namespace Rcpp;
 //' @param mcov Matrix. The Sigma-matrix in the Mahalanobis distance formula (\code{stats::cov(Xtrain_mat)}) gives Mahalanobis distance,
 //' \code{diag(m)} gives the Euclidean distance.
 //' @param S_scale_dist Logical indicating
+//' @param Xtrain_mat Matrix
+//' @param Xtest_mat Matrix
```
Comment (martinju) [inline src/distance.cpp]: Document properly

**Comment by nikolase90** [comment]
Fixed spelling errors with 431e831.

**Comment by martinju** [comment]
> > Note done yet, but here are the first half of the review
> > A few general comments in addition to the inline comments below:
> > 
> > * [ ]   I think we should run some spell checking. See for instance this package: https://github.com/ropensci/spelling. It also gives ability to run the spell checker as a unit test.
> > * [ ]   I suggest we change `m` and `nfeatures` to `n_features` for consistency throughout the package.
> > * [ ]   We should be better at specifying what is the default value of input parameters. At least I find that helpful  when reading documentations.
> 
> Agree to most of it. I'll do the spell check, but I think we should only add documentation changes in this PR. So we could streamline the naming of the variables in a different one (just to make sure that the actually code changes is clear).
> 
> It is not necessary to specify the default parameters when documenting the actual parameters. The default values are shown in the `Usage` section, i.e. run `?shapr` to see an example.

Ok to do name changes in another PR.
I know it is not necessary to document the default, but if you have a lot of parameters I find it helpful not have to scroll to look for the default value. I am not saying we should do it for all parameters, but I think it should be done for some. I can go through it in a separate PR next week once this is done.

File: R/predictions.R [hunk 19]
```diff
@@ -1,17 +1,32 @@
...
 #'
-#' @description
-#' TODO: Write a better description
+#' @description This function should only be called internally, and not used as
+#' a stand-alone function.
 #'
 #' @param dt data.table
-#' @param prediction_zero Positive integer between 0 & 1.
-#' @param explainer An object of class \code{explainer}
+#' @param prediction_zero Numeric
```
[the remark this answers is not in the document]
Comment (martinju) [reply R/predictions.R]:
```suggestion
#' @param prediction_zero Numeric. The value to use for phi_0.
```

File: R/explanation.R (hunk 11 above)
[the remark this answers is not in the document]
Comment (martinju) [reply R/explanation.R]: You are right. You can then probably ignore other similar comments.

File: R/features.R [hunk 20]
```diff
@@ -1,24 +1,41 @@
-#' Get combinations
+#' Define feature combinations, and fetch additional information about each unique combination
 #'
-#' @inheritParams global_arguments
+#' @param m Positive integer. Total number of features.
```
[the remark this answers is not in the document]
Comment (nikolase90) [reply R/features.R]: Will address this is a different PR. See #https://github.com/NorskRegnesentral/shapr/issues/136.

File: R/features.R (hunk 2 above)
Comment (nikolase90) [reply R/features.R]: Will address this is a different PR. See #136.

File: R/explanation.R (hunk 11 above)
[the remark this answers is not in the document]
Comment (nikolase90) [reply R/explanation.R]: Goodie. Will the names in a different PR. See #136.

File: R/models.R (hunk 5 above)
Comment (martinju) [reply R/models.R]: You are right. I copied this from the documentation in predict_model. I think we should have it there though, along with the combination approach, under a section called "Advanced usage" or something. Agree? If so, I can open a PR and do it. There is a link there that needs to be fixed anyway.

File: R/features.R (hunk 1 above)
[the remark this answers is not in the document]
Comment (nikolase90) [reply R/features.R]: Do you mean that we should remove the argument entirely?

File: R/models.R (hunk 5 above)
Comment (nikolase90) [reply R/models.R]: Sounds good!

File: R/plot.R [hunk 21]
```diff
@@ -1,15 +1,20 @@
...
 #' @param digits Integer. Number of significant digits to use in the feature description
-#' @param plot_phi0 Logical. Whether to include phi0 in the plot
-#' @param index_x_test Integer vector. Which of the test observations to plot
-#' @param top_k_features Integer. How many features to include in the plot
+#' @param plot_phi0 Logical. Whether to include \code{phi0} in the plot
+#' @param index_x_test Integer vector. Which of the test observations to plot. E.g. if you have
+#' explained 10 observations using \code{\link{explain}}, you can generate a plot for the first 5
+#' observations by setting \code{index_x_test = 1:5}.
+#' @param top_k_features Integer. How many features to include in the plot. E.g. if you have 15
+#' features in your model you can plot the 5 most important features by setting
```
[the remark this answers is not in the document]
Comment (nikolase90) [reply R/plot.R]: Fixed with aa8abbe.

File: R/predictions.R [hunk 22]
```diff
@@ -1,17 +1,32 @@
...
 #'
-#' @description
-#' TODO: Write a better description
+#' @description This function should only be called internally, and not be used as
+#' a stand-alone function.
 #'
 #' @param dt data.table
-#' @param prediction_zero Positive integer between 0 & 1.
-#' @param explainer An object of class \code{explainer}
+#' @param prediction_zero Numeric
```
Comment (nikolase90) [inline R/predictions.R]:
```suggestion
#' @param prediction_zero Numeric. The value to use for \code{phi_0}.
```

File: R/sampling.R [hunk 23]
```diff
@@ -1,20 +1,28 @@
...
 #'
-#' @param index_given Vector. The indices of the features to condition upon.
+#' @param index_given Integer vector. The indices of the features to condition upon. Note that
+#' \code{min(index_given) >= 1} and \code{max(index_given) <= p}.
 #' @param p Positive integer. The number of features.
-#' @param x_test_gaussian Vector with the Gaussian transformed features of the observation whose
-#' predictions ought to be explained (test data). Dimension \code{1xp} or \code{px1}.
-#' @param x_test Matrix, data.frame or data.table with the features of the observation whose
-#' predictions ought to be explained (test data). Dimension \code{1xp} or \code{px1}.
+#' @param x_test_gaussian Numeric matrix. Contains the Gaussian transformed features of the
```
[the remark this answers is not in the document]
Comment (nikolase90) [reply R/sampling.R]: Fixed with e8119b8.

File: R/transformation.R (hunk 15 above)
Comment (nikolase90) [reply R/transformation.R]: Fixed with 92b9829074397c99f5e305f8dc7652a2314cfd43.

File: R/sampling.R [hunk 24]
```diff
@@ -1,20 +1,28 @@
 #' Sample conditional variables using the Gaussian copula approach
 #'
-#' @param index_given Vector. The indices of the features to condition upon.
+#' @param index_given Integer vector. The indices of the features to condition upon. Note that
+#' \code{min(index_given) >= 1} and \code{max(index_given) <= p}.
 #' @param p Positive integer. The number of features.
```
[the remark this answers is not in the document]
Comment (nikolase90) [reply R/sampling.R]: Haha! Updated #136!

File: R/shapley.R [hunk 25]
```diff
@@ -16,10 +20,10 @@ shapley_weights <- function(m, N, s, weight_zero_m = 10^6) {
 #' Calculate weighted matrix
 #'
 #' @param X data.table
+#' @param use_shapley_weights_in_W Logical
+#' @param normalize_W_weights Logical
```
[the remark this answers is not in the document]
Comment (nikolase90) [reply R/shapley.R]: Help me out? Not sure about those.

File: R/transformation.R (hunk 14 above)
Comment (nikolase90) [reply R/transformation.R]: We could just remove the argument, and use the default value? I don't think we should use `... `, since it could mess with the other arguments in `quantile` as well.

File: R/zzz.R (hunk 16 above)
Comment (nikolase90) [reply R/zzz.R]: Fixed in #127.

File: R/shapley.R (hunk 12 above)
Comment (nikolase90) [reply R/shapley.R]: Fixed with 06a2981.

File: R/shapley.R [hunk 26]
```diff
@@ -44,28 +48,75 @@ weight_matrix <- function(X, use_shapley_weights_in_W = TRUE, normalize_W_weight
...
+#' i.e. the explanatory variables. Note that the response variable should not be part of
+#' \code{x}, and that \code{ncol(x) = p} equals the total number of features.
 #'
-#' @param model The model whose predictions we want to explain.
+#' @param model The model whose predictions we want to explain. See \code{\link{predict_model}}
+#' for more information about which models \code{shapr} supports.
 #'
 #' @param n_combinations Integer. The number of feature combinations to sample. If \code{NULL},
 #' the exact method is used and all combinations are considered. The maximum number of
 #' combinations equals \code{2^p}.
```
[the remark this answers is not in the document]
Comment (nikolase90) [reply R/shapley.R]: Fixed with 06a2981.

File: R/shapley.R [hunk 27]
```diff
@@ -1,11 +1,15 @@
 #' Calculate Shapley weight
 #'
-#' @inheritParams global_arguments
+#' @param m Positive integer. Total number of features.
+#' @param s Positive integer. Represents the number of features you want to sample from a feature
+#' space consisting of \code{m} unique features. Note that \code{s>= 0} and \code{s <= m}.
+#' @param N Positive integer. The number of unique combinations when sampling \code{s} features,
+#' without replacement, from a sample space consisting of \code{m} different features.
+#' @param weight_zero_m Positive integer. Represents the Shapley weigth for two special
```
[the remark this answers is not in the document]
Comment (nikolase90) [reply R/shapley.R]: Agree. Since that requires some code changes I opened https://github.com/NorskRegnesentral/shapr/issues/137.

File: R/features.R (hunk 1 above)
[the remark this answers is not in the document]
Comment (martinju) [reply R/features.R]: Yes (if you cannot see any use cases for this either).

[the remark this answers is not in the document]
Comment (martinju) [reply R/features.R]: And include a good comment directly in the code.

File: R/transformation.R (hunk 14 above)
Comment (martinju) [reply R/transformation.R]: Ok, let's remove it. It's only used internally anyway, so nobody will ever touch it.

File: R/shapley.R (hunk 25 above)
Comment (martinju) [inline R/shapley.R]:
```suggestion
#' @param normalize_W_weights Logical. Whether to normalize the weights for the combinations to sum to 1 for increased numerical stability before solving the WLS (weighted least squares). Applies to all combinations except combination \code{1} and \code{2^m}.
```

File: R/shapley.R [hunk 28]
```diff
@@ -16,10 +20,10 @@ shapley_weights <- function(m, N, s, weight_zero_m = 10^6) {
 #' Calculate weighted matrix
 #'
 #' @param X data.table
+#' @param use_shapley_weights_in_W Logical
```
Comment (martinju) [inline R/shapley.R]: We should probably do this in a different PR, but to me, it seems that we don't need this argument after we remove the reduce_dim argument. We should then always just use the shapley weights without as the w-vector (i.e. without multiplying with the no column), as the only situation where no can be different from 1 is when reduce_dim = FALSE.

File: R/shapley.R (hunk 25 above)
[the remark this answers is not in the document]
Comment (martinju) [reply R/shapley.R]: Yep, see the inline comments.

File: R/shapley.R (hunk 28 above)
Comment (martinju) [reply R/shapley.R]: Anyway, for now, I suggest
```suggestion
#' @param use_shapley_weights_in_W Logical. Whether to use direct formula based (TRUE) or sampling based (FALSE) weights for the combinations.
```

File: src/AICc.cpp (hunk 17 above)
Comment (nikolase90) [reply src/AICc.cpp]: `*.cpp` will be done separately.

File: src/distance.cpp (hunk 18 above)
Comment (nikolase90) [reply src/distance.cpp]: *.cpp will be done in a different PR.

File: R/transformation.R (hunk 14 above)
Comment (nikolase90) [reply R/transformation.R]: Added to #138 .

File: R/features.R (hunk 1 above)
[the remark this answers is not in the document]
Comment (nikolase90) [reply R/features.R]: Added in #138

**Comment by nikolase90** [comment]
@martinju Then I think we have fixed everything?

File: R/clustering.R [hunk 29]
```diff
@@ -1,9 +1,16 @@
...
+#' @param cor_mat Numeric matrix. Note that \code{x_mat} or \code{cor_mat} must be included,
+#' otherwise the function will return an error message.
+#' @param alpha Numeric. Weight for number of clusters. See \code{\link[maptree]{kgs}}.
+#'
+#' @details The returned list contains the following items
+#' \describe{
+#'   \item{cor_mat}{TODO: Add information}
+#'   \item{which_k}{TODO: Add information}
+#'   \item{cluster}{TODO: Add information}
+#' }
```
Comment (martinju) [inline R/clustering.R]: Some TODOs here, but don't think we need to do that now.

Comment (nikolase90) [reply R/clustering.R]: Yeah. Let's skip that now.
